torch/_decomp/decompositions.py

```diff
@@ -3434,6 +3445,54 @@ def arange_start(
     )
 
 
+def linalg_solve_is_vector_rhs(input: Tensor, other: Tensor) -> bool:
+    expected_batched_rhs_shape = input.shape[:-1]
+    vector_case = other.ndim == 1 or (
+        input.ndim - 1 == other.ndim and other.shape == expected_batched_rhs_shape
+    )
+    return vector_case
+
+
+@register_decomposition([aten._linalg_solve_ex.default, aten._linalg_solve_ex.result])
+@out_wrapper("result", "LU", "pivots", "info")
+def _linalg_solve_ex(
+    A: Tensor, B: Tensor, *, left: bool = True, check_errors: bool = False
+) -> Tuple[Tensor, Tensor, Tensor, Tensor]:
+    checkFloatingOrComplex(A, "linalg.solve")
+    utils.check(
+        A.dtype == B.dtype,
+        lambda: (
+            f"linalg.solve: Expected A and B to have the same dtype, but found A of type "
+            f"{A.dtype} and B of type {B.dtype} instead"
+        ),
+    )
+    vector_case = linalg_solve_is_vector_rhs(A, B)
+    B_ = B.unsqueeze(-1) if vector_case else B
+    checkInputsSolver(A, B_, left, "linalg.solve")
+    B_broad_shape, _ = _linalg_broadcast_batch_dims(B_, A)
+    utils.check(
+        left or not vector_case,
+        lambda: (
+            "linalg.solve: Vector broadcasting of the left hand side is not supported for left=False. "
+            "In this case linalg.solve is equivalent to B / A.squeeze(-1)"
+        ),
+    )
+    use_A_T = A.is_contiguous() and not A.is_complex()
+    LU, pivots, info = torch.linalg.lu_factor_ex(A.mT if use_A_T else A)
+    if check_errors:
+        aten._linalg_check_errors(info, "torch.linalg.solve_ex", is_matrix=A.ndim == 2)
+    result_ = torch.linalg.lu_solve(LU, pivots, B_, left=left, adjoint=use_A_T)
+    result_shape = B_broad_shape[:-1] if vector_case else B_broad_shape
+    result = torch.empty_strided(
+        size=result_shape,
+        stride=make_contiguous_strides_for(result_shape, not left),
+        dtype=B.dtype,
+        device=B.device,
+    )
+    result.copy_(result_.reshape(result_shape))
```

Comment:
`empty_strided` and `copy_` are needed to get strides right. On the CPP side, it's not exactly like that. Instead, result is allocated first with the proper strides in the meta, which is passed to impl.

Reply:
Is it possible to just call `.contiguous()` on result instead of an in-place modification here?

Reply:
Unfortunately, I cannot think of any other way to do it. But open to other ideas.

Reply:
Why doesn't just doing `result = result_.contiguous()` suffice here?

Reply:
I don't think that that will work, as the result is not contiguos but fortran contig.

That being said, please don't follow the implementation on eager line by line, as it's not composite compliant, and here we want something that's out-of-place.

Reply:
Made another attempt to fix this, no luck.

How about going back to the original PR that just added a meta? I think it passed all the tests. In the meta, I can allocate the result the same way as on the CPP side because I don't need to do compute there.

With the decomp, I need to do compute and need to get the strides correct. And I see no way of doing it. Either I can do `result_.reshape(result_shape)` after calling `lu_solve` to get correct numerical output; OR I can call something like `result_.as_strided(result_shape, make_contiguous_strides_for(result_shape, not left))` to get the correct strides, but the output will be wrong. Both of these are without calling `empty_strided` and doing the copy.

I guess the issue with the decomp is due to the CPP code specifying the strides of the result in the meta. While with the inplace ops in the decomp, we have no way to control the strides of the result buffer directly.

Thoughts?